web-console/developer/topology/topology.md

```diff
@@ -3,5 +3,276 @@ The Topology page is a primary nav item in the OpenShift Developer perspective.
 
+## Topology
+The Topology view would be used in OpenShift Dev-Console to show semantic aggregated metadata about related pods, deployments, builds of application components in a namespace.
+
+![Topology View](img/Topology_View.png)
+
+
+### No Project in Context
+When no project is selected, Topology View will prompt users to select from a given list of projects.
+
+![No project in context](img/no_project_exists.png)
+
+
+#### No Workload Exists
+When no workload exists, the topology view redirects users to the add flows to start creating new workloads to be shown on the topology canvas.
+![Topology View](img/no_project_exists.png)
+
+### Application selector
+Users could choose to select either one of the applications or all the applications from the application selector.
+
+## Topology View
+
+
+### 1. Navigation item:
+
+### 2. Masthead
+The masthead in topology view includes the project and application selector
+
+### 3. Topology Canvas
+This is the area where the workloads are visually placed in the Topology view.
+
+### 4. Toolbar
+The toolbar consists of the following controls:
+Zoom-in
+Zoom-out
+Fit to Screen
+Reset View
+
+### 5. Side Panel
+On selecting any workload, connectors or application group on Topology View, their respective details appear on a sliding panel on the right hand side.
+
+### 6. Nodes
+![Node design](img/node_design.png)
+
+Resources of type Deployments, Deployment Configs & xxx are shown as circle nodes in the topology view.
+
+The circle node includes a pod status indicator, an image, 4 decorators and a node label
+
+The pod donut is segmented by status. Between each segment there’s a space of 3px.
+
+The name label contains a badge before the name, that denotes the type of resource.
+
+
+#### Pod Status Indicator
+On hovering over the pod donut, the information for that particular segment appears in an info tip.
+
+![Pod Status](img/pod_status.png)
+
+
+#### Node Image
+The node image consists of the build image for the resource.
+
+#### Decorators
+
+##### Route Decorator
+The route decorator is located on the top right quadrant of the node.
+Go To External Link
+
+##### Code Decorator
+The code decorator is located on the bottom right quadrant of the node.
+Go To GitHub or CHE Workspace
+
+![Pod Status](img/che_icon.png)    If CHE workspace is available, CHE icon should be used
+
+![Pod Status](img/github_icon.png)
+If not, determine	the git client associated and use that icon
+
+![Pod Status](img/git_alt.png)  If git couldn’t be determined, use git-alt icon from font-awesome
+
+
+##### Build Decorator
+The build decorator in the topology view is located in the bottom left of the node.  The decorator has six options: new, pending, running, completed, failed, and cancelled. When the decorator is hovered on, the status of the build will appear.  When the decorator is clicked, the user will be taken to the overview tab of the “Builds” details page.
```

Comment:
The build decorator provides a one click action to view the build logs. (the last sentence is incorrect)